shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -0,0 +1,321 @@
+/// Only Cassandra queries to the `system.peers` table found via the `is_system_peers(_v2)` functions should be passed to this
+fn rewrite_port(message: &mut Message, new_port: u32) {
+    if let Frame::Cassandra(frame) = &mut message.original {
+        if let CassandraOperation::Result(CassandraResult::Rows { value, metadata }) =
+            &mut frame.operation
+        {
+            let port_column_index = metadata
+                .col_specs
+                .iter()
+                .position(|col| col.name.as_str() == "native_port");
+
+            if let Some(i) = port_column_index {
+                if let MessageValue::Rows(rows) = &mut *value {
+                    for row in rows.iter_mut() {
+                        row[i] = MessageValue::Integer(new_port as i64, IntSize::I32);
+                    }
+                }
+            }
+        } else {
+            panic!(
+                "Expected CassandraOperation::Result(CassandraResult::Rows), got {:?}",
+                frame
+            );
+        }
+    } else {
+        panic!("Expected Frame::Cassandra, got {:?}", message.original);
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::frame::CassandraFrame;
+    use crate::message::MessageDetails;
+    use crate::transforms::cassandra::peers_rewrite::CassandraResult::Rows;
+    use cassandra_protocol::{
+        compression::Compression,
+        consistency::Consistency,
+        frame::{
+            frame_result::{
+                ColSpec,
+                ColType::{Inet, Int},
+                ColTypeOption, RowsMetadata, RowsMetadataFlags, TableSpec,
+            },
+            Direction, Frame as CassandraProtocolFrame, Opcode, Serialize,
+            {frame_query::BodyReqQuery, Flags, Version},
+        },
+        query::QueryParams,
+    };
+
+    fn create_query_message(query: String) -> Message {
+        let body = BodyReqQuery {
+            query,
+            query_params: QueryParams {
+                consistency: Consistency::One,
+                with_names: false,
+                values: None,
+                page_size: Some(5000),
+                paging_state: None,
+                serial_consistency: None,
+                timestamp: Some(1643855761086585),
+            },
+        };
+
+        let frame = CassandraProtocolFrame {
+            version: Version::V4,
+            direction: Direction::Request,
+            flags: Flags::default(),
+            stream_id: 0,
+            opcode: Opcode::Query,
+            tracing_id: None,
+            body: body.serialize_to_vec(),
+            warnings: vec![],
+        };
+
+        let bytes = frame.encode_with(Compression::None).unwrap();
+
+        Message {
+            details: MessageDetails::Unknown,
+            return_to_sender: false,
+            modified: false,
+            original: Frame::Cassandra(CassandraFrame::from_bytes(bytes.into()).unwrap()),
```

Comment:
Can we just directly use a CassandraFrame here without round tripping through bytes?